File: 프로그래머스/lv2/42885. 구명보트/구명보트.java

import java.util.Arrays;
class Solution {
    public int solution(int[] people, int limit) {
       	int answer = 0;
        int largest = people.length-1;
		int smallest=0;
		
		Arrays.sort(people);

		while(smallest<=largest) {
			if(people[smallest]+people[largest] <= limit) {
				smallest++;
			}
			answer++;
			largest--;
		}
        
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 0
    big_idx = len(people) - 1
    small_idx = 0
    
    people.sort()

    while small_idx <= big_idx:
        if people[small_idx] + people[big_idx] <= limit:
            small_idx = small_idx + 1
            
        answer=answer+1
        big_idx=big_idx-1
    
    return answer